pydata_center/train.py

```diff
@@ -0,0 +1,142 @@
+import argparse
+import datetime
+import json
+import os
+
+import django
+import numpy as np
+from data.dataset import build_datasets, fetch_raw_metrics
+from django.conf import settings
+from predictive.forecasting import (train_anomaly_model,
+                                    train_classification_model,
+                                    train_regression_model)
+from predictive.models.model_io import save_model
+
+from pydata_center.utils.logger import setup_logger
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'pydata_center.settings')
+django.setup()
+
+logger = setup_logger()
+
+AVAILABLE_MODELS = ['regression', 'anomaly', 'classifier']
+
+
+def prepare_model_dir(output_dir: str | None) -> str:
+    """
+    Determine where to write model files.
+    If `output_dir` is provided, use that; otherwise
+    default to BASE_DIR/predictive/models.
+    """
+    if output_dir:
+        model_dir = output_dir
+    else:
+        model_dir = os.path.join(settings.BASE_DIR, 'predictive', 'models')
+    os.makedirs(model_dir, exist_ok=True)
+    return model_dir
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description='Train predictive models')
+    parser.add_argument(
+        '--days', type=int, default=7,
+        help='How many days of metrics to use'
+    )
+    parser.add_argument(
+        '--window', type=int, default=5,
+        help='Sliding window size'
+    )
+    parser.add_argument(
+        '--contamination', type=float, default=0.05,
+        help='IsolationForest contamination level'
+    )
+    parser.add_argument(
+        '--test-size', type=float, default=0.2,
+        help='Test split ratio'
+    )
+    parser.add_argument(
+        '--metrics-out', type=str, default=None,
+        help='Optional path to save training metrics JSON'
+    )
+    parser.add_argument(
+        '--model-types', nargs='*', default=AVAILABLE_MODELS,
+        choices=AVAILABLE_MODELS,
+        help='Which models to train'
+    )
+    parser.add_argument(
+        '--output-dir', type=str, default=None,
+        help='Where to save trained model files'
+    )
+    return parser.parse_args()
+
+
+def train_models(
+    X_reg: np.ndarray, y_reg: np.ndarray,
+    X_clf: np.ndarray, y_clf: np.ndarray,
+    contamination: float, test_size: float,
+    model_types: list[str]
+) -> dict[str, tuple]:
+    """
+    Train only the requested model types and return a dict:
+      {
+        'regression': (model, metrics),
+        'anomaly':    (model, {}),
+        'classifier': (model, metrics),
+      }
+    """
+    results: dict[str, tuple] = {}
+
```

Comment:
Before training, add checks/logging for data sufficiency; fail gracefully if insufficient data is present.